mailbox/api/src/main/java/org/apache/james/mailbox/model/Attachment.java

```diff
@@ -73,19 +77,57 @@ private AttachmentId attachmentId() {
             }
             return AttachmentId.random();
         }
+    }
+
+    public static class WithBytes {
+        private final byte[] bytes;
+        private final Attachment metadata;
+
+        public WithBytes(byte[] bytes, Attachment metadata) {
+            Preconditions.checkArgument(bytes != null);
+            this.bytes = bytes;
+            this.metadata = metadata;
+        }
+
+        public Attachment getMetadata() {
+            return metadata;
+        }
+
+        public ByteArrayInputStream getStream() {
+            return new ByteArrayInputStream(bytes);
+        }
+
+        /**
+         * Be careful the returned array is not a copy of the attachment byte array.
+         * Mutating it will mutate the attachment!
+         * @return the attachment content
+         */
+        public byte[] getBytes() {
```

Comment:
why do we provide this method if it's dangerous?

Reply:
Because... We don't want to do a full copy of it, and we made the choice in the past that it was a trade-off woth taking.

I don't intend to change that.